code/modules/mob/living/carbon/xenomorph/damage_procs.dm

```diff
@@ -121,6 +121,13 @@
 
 	damage = process_rage_damage(damage, src)
 
+	if(overhealth < damage)
+		damage -= overhealth
+		overhealth = 0
+	else
+		overhealth -= damage
+		damage = 0
```

Comment:
```suggestion
		damage = 0
		updatehealth()
		return TRUE
```

Reply:
I figured updatehealth() needs to be called also.

Reply:
What for? There wasn't any change in health.